### PR TITLE
Build with Flatpak

### LINKED ISSUE #301: Package as Flatpak for elementary OS 6
As you might have heard, AppCenter is going Flatpak! For elementary OS 6, we will not longer be accepting Debian packaging, so your app will need to be updated.

We have documentation on our developer guide for writing your Flatpak manifest here: https://docs.elementary.io/develop/writing-apps/our-first-app/packaging#flatpak-manifest

Also, in case you missed it, elementary OS 6 beta 2 is [now available](https://builds.elementary.io). This is a great time to get your app packaged up as Flatpak and test that everything is working before the final release. AppCenter submission for OS 6 will be open soon, so don't delay :)

### PR DESCRIPTION
**Pull request by danrabbit**
Fixes #301 

Todo:
- [x] disable tests in CI
- [x] fix up with meson 
- [x] Figure out what to do with `Not exporting share/icons/hicolor/64x64/mimetypes/application-x-spiceup.svg, non-allowed export filename`

### CODE REVIEW

File: com.github.philip_scott.spice-up.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  - '--share=ipc'
+  - '--socket=fallback-x11'
+  - '--socket=wayland'
+
+  - '--device=all'      # Required for Controllers
+  - '--filesystem=home' # Required to read/write files
+  - '--talk-name=org.freedesktop.ScreenSaver' # Required to stop screen from sleeping
+
+modules:
+  - name: gudev
```
Comment (Philip-Scott) [inline com.github.philip_scott.spice-up.yml:19]: Where did you get all of these module info? is there a simpler way to go from the local `*-dev` dependency to what needs to be written on the flatpak yaml? This was honestly one of my the big pain points when doing the other migrations :sweat_smile:

Comment (danrabbit) [reply com.github.philip_scott.spice-up.yml:19]: TBH I did a GitHub search and copied what other people had 😅

File: com.github.philip_scott.spice-up.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ finish-args:
   - '--socket=wayland'
 
   - '--device=all'      # Required for Controllers
-  - '--filesystem=home' # Required to read/write files
+  - '--filesystem=xdg-documents' # Required for library
```
Comment (Philip-Scott) [inline com.github.philip_scott.spice-up.yml:15]: Would this limit the library access to only files on the Documents folder? :thinking:

Comment (danrabbit) [reply com.github.philip_scott.spice-up.yml:15]: The open/save dialog uses the portal, so you can always open or save files from anywhere manually and Flatpak persists those permissions in the document portal. This just makes it so those presentations that you create from the welcome screen can be saved to the documents folder

**Comment by jtagcat** [comment]
Will this reach https://github.com/flathub/com.github.philip_scott.spice-up, to be distributed to the users?

**Pull request merged**